assets/app/scripts/controllers/overview.js

```diff
@@ -386,7 +418,106 @@ angular.module('openshiftConsole')
       });
     });
 
+    var updateTimeout = null;
+
+    function updateTopology() {
+      updateTimeout = null;
+
+      var topologyRelations = [];
+      var topologyItems = { };
+
+      // Because metadata.uid is not unique among resources
+      function makeId(resource) {
+        return resource.kind + resource.metadata.uid;
+      }
+
+      // We only add pods that are pointed to by a service
+      angular.forEach($scope.pods, function(pod) {
+        if (showMonopod(pod)) {
+          topologyItems[makeId(pod)] = pod;
+        }
+      });
+
+      // Put each of the resources into the topology. Each of these
+      // are name -> object mappings.
+      [ $scope.routes,
+        $scope.services,
+        $scope.deployments,
+        $scope.deploymentConfigs
+      ].forEach(function(resources) {
+        angular.forEach(resources, function(resource) {
+          topologyItems[makeId(resource)] = resource;
+        });
+      });
+
+      /*
+       * Most of the tables in this scope are in a standard form with
+       * string keys, pointing to a map of further name -> object mappings.
+       *
+       * We can process them all together.
+       *
+       * Only relationships for objects added above will be displayed by
+       * the topology. So we can (for example) leave relations for monopods in.
+       */
+      [
+        [ $scope.podsByService, $scope.services ],
+        [ $scope.monopodsByService, $scope.services ],
+        [ $scope.podsByDeployment, $scope.deployments ],
+        [ $scope.deploymentsByService, $scope.services ],
+        [ $scope.deploymentConfigsByService, $scope.services ],
+        [ $scope.routesByService, $scope.services ]
+      ].forEach(function(args) {
+        var map = args[0];
+        var lookup = args[1];
+        angular.forEach(map, function(resources, name) {
+          var source = lookup[name];
+          if (source) {
+            angular.forEach(resources, function(resource) {
+              topologyRelations.push({ source: makeId(source), target: makeId(resource) });
+            });
+          }
+        });
+      });
+
+      $scope.topologyItems = topologyItems;
+      $scope.topologyRelations = topologyRelations;
+      $scope.$digest();
+    }
+
+    function updateTopologyLater() {
+      if (!updateTimeout) {
+        updateTimeout = window.setTimeout(updateTopology, 100);
+      }
+    }
+
+    $scope.$on("select", function(ev, resource) {
```

Comment:
The topology-graph component. See its documentation: https://github.com/kubernetes-ui/topology-graph#select